test/core/QCheck_expect_test.ml

```diff
@@ -127,6 +136,70 @@ module Generator = struct
       ~name:"tree_rev_is_involutive"
       QCheck.(make IntTree.gen_tree)
       (fun tree -> IntTree.(rev_tree (rev_tree tree)) = tree)
+
+  let nat_split2_spec =
+    Test.make ~name:"nat_split2 spec"
+      (make
+        ~print:Print.(pair int (pair int int))
+         Gen.(small_nat >>= fun n ->
+              pair (return n) (nat_split2 n)))
+      (fun (n, (a, b)) ->
+         0 <= a && 0 <= b && a + b = n)
+
+  let pos_split2_spec =
+    Test.make ~name:"pos_split2 spec"
+      (make
+        ~print:Print.(pair int (pair int int))
+         Gen.(small_nat >>= fun n ->
+              (* we need n > 2 *)
+              let n = n + 2 in
+              pair (return n) (pos_split2 n)))
+      (fun (n, (a, b)) ->
+         (0 < a && 0 < b && a + b = n))
+
+  let range_subset_spec =
```

Comment:
Since `range_subset` comes with a "distribution spec", would it make sense to supplement with a stat expect test? "Many repetitions of `~size:1` distribute evenly in a histogram", "for a single run with large `size` and larger range the avg/median/variance come out as expected", ...? (just a thought)

Reply:
I want to avoid probabilistic tests that fail with non-zero probability.

(Note: I wrote a "reference implementation" of these functions [in random-generator](https://github.com/gasche/random-generator/commit/74c2ea0f57d79c56328e6c6d6bf6eb2850436153), a small generator library that is parametrized over a probability monad, and can in particular compute discrete distributions. There one can check uniformity for small inputs, and I did. The fact that I completely messed up the port to QCheck, introducing several bugs, suggests that this testing strategy is not enough...)

Reply:
I agree about avoiding introducing CI-tests that fail with a non-zero probability.
Currently, CI-tests are run with the same seed however, which is deterministic (if not optimal).
By "stat tests" I just meant a test in `module Stats`.

While writing tuple/bind tests a few months back I added a couple of stat tests of these to confirm visually - which works nicely with expect tests: https://github.com/c-cube/qcheck/pull/176

For example, below is a (pretty IMHO) histogram and basic (deterministic 😄) average/stddev/... of the distribution of the sum of two `int`s in [0;100]. It is not a great guarantee, but still nice to confirm a reasonable average and histogram shape and the expected min and max:
https://github.com/c-cube/qcheck/blob/654abe135b21154e94e8690e0ac2b6a19e8888c8/test/core/QCheck_expect_test.expected#L507-L531

Reply:
I added a simple `Stats` test for `size:1` to visualize the distribution in 71675b5276372310508885986d5c41bafddce094